spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -1038,6 +1038,26 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
       }
     }
   }
+  test("hex") {
+    Seq(true, false).foreach { dictionaryEnabled =>
+      withTempDir { dir =>
+        val path = new Path(dir.toURI.toString, "hex.parquet")
+        makeParquetFileAllTypes(path, dictionaryEnabled = dictionaryEnabled, 10000)
+
+        withParquetTable(path.toString, "tbl") {
+          // ints
+          checkSparkAnswerAndOperator("SELECT hex(_2), hex(_3), hex(_4), hex(_5) FROM tbl")
+
+          // uints, uint8 and uint16 not working yet
+          // checkSparkAnswerAndOperator("SELECT hex(_9), hex(_10), hex(_11), hex(_12) FROM tbl")
+          checkSparkAnswerAndOperator("SELECT hex(_11), hex(_12) FROM tbl")
+
+          // strings, binary
+          checkSparkAnswerAndOperator("SELECT hex(_8), hex(_14) FROM tbl")
```

Comment:
What about other types like _1, _6, _7, _13, _15+?

Reply:
1caa6fd adds the other columns... `_13` also fails failure.

Reply:
I updated dictionary handling, so I've added _13 back in. I think only the smaller uints are omitted now.